Src/AutoFixtureUnitTest/Utf8EncodingGeneratorTest.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Ploeh.AutoFixture;
+using Ploeh.AutoFixture.Kernel;
+using Ploeh.AutoFixtureUnitTest.Kernel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace Ploeh.AutoFixtureUnitTest
+{
+    public class Utf8EncodingGeneratorTest
+    {
+        [Fact]
+        public void SutIsSpecimenBuilder()
+        {
+            var sut = new Utf8EncodingGenerator();
+            Assert.IsAssignableFrom<ISpecimenBuilder>(sut);
+        }
+
+        [Fact]
+        public void CreateWithEncodingRequestWillReturnUtf8Encoding()
+        {
+            // Arrange
+            var sut = new Utf8EncodingGenerator();
+
+            // Act
+            var result = sut.Create(typeof(Encoding), new DelegatingSpecimenContext());
+
+            // Assert
+            Assert.Equal(Encoding.UTF8, result);
+        }
+
+        [Fact]
+        public void CreateWithNullRequestWillReturnNoSpecimen()
+        {
+            // Arrange
+            var sut = new Utf8EncodingGenerator();
+
+            // Act
+            var result = sut.Create(null, new DelegatingSpecimenContext());
+
+            // Assert
+            Assert.Equal(new NoSpecimen(), result);
+        }
+
+        [Fact]
+        public void CreateWithNullContextDoesNotThrow()
```

Comment:
> Would you say that that is a violation of the open/closed principle?

You could argue that. In the case of AutoFixture (and, I suppose, many other open source libraries), the reason is more practical, though. The tests constitute the public contract that we (the maintainers) swear to uphold.

As a general rule, as long as one can fix a bug or introduce a new feature without breaking any of the existing tests, we regard that as a non-breaking change. If we need to change tests, we regard it as a breaking change.

This is important because AutoFixture explicitly follows semantic versioning.

This would all be perfect if it was true, but alas it isn't. There are some tests (particularly `Ploeh.AutoFixtureUnitTest.DefaultPrimitiveBuildersTest.SutHasCorrectContents` - we really ought to delete that test) that tend to change on a regular basis, but at least these are well-known exceptions to the rule.

For that reason, if I can predict in advance that a test might be one that we might want to change in the future, that's a sign to me that perhaps it only tests an implementation detail. If that's the case, I'd prefer to not have it in the first place.